src/scripts/main.js

```diff
@@ -1,3 +1,388 @@
+
+function clickOnTr(event) {
+  const currentTr = event.target.closest('tr');
+
+  if (!currentTr) {
+    return;
+  }
+
+  if (currentTr.classList.contains('active')) {
+    currentTr.classList.remove('active');
+
+    return;
+  }
+
+  const allRows = document.querySelectorAll('tbody tr');
+
+  allRows.forEach((row) => row.classList.remove('active'));
+
+  currentTr.classList.add('active');
+}
+
+tbody.addEventListener('click', clickOnTr);
+
+// #endregion active-tr
+
+// #region add form
+
+const body = document.body;
+const newForm = document.createElement('form');
+
+newForm.classList.add('new-employee-form');
+
+body.append(newForm);
+
+newForm.innerHTML = `
+    <label>Name: <input required data-qa="name" name="name" type="text"></label><br>
+    <label>Position: <input required data-qa="position" name="position" type="text"></label><br>
+    <label>Office: <select required name="office" data-qa="office">
+      <option value="Tokyo">Tokyo</option>
+      <option value="Singapore">Singapore</option>
+      <option value="London">London</option>
+      <option value="New York">New York</option>
+      <option value="Edinburgh">Edinburgh</option>
+      <option value="San Francisco">San Francisco</option>
+    </select></label><br>
+    <label>Age: <input required data-qa="age" name="age" type="number" step="1" min="18" max="90"></label><br>
+    <label>Salary: <input required data-qa="salary" name="salary" type="text"></label><br>
+    <button type="submit">Save to table</button>
+`;
+
+newForm.style.zIndex = '0';
+const formButton = newForm.querySelector('button');
+const employeeForm = document.querySelector('.new-employee-form');
+
+// eslint-disable-next-line no-shadow
+function clickOnButton(event) {
+  event.preventDefault();
+
+  const nameInput = employeeForm.querySelector('[data-qa="name"]');
+  const positionInput = employeeForm.querySelector('[data-qa="position"]');
+  const officeInput = employeeForm.querySelector('[data-qa="office"]');
+  const ageInput = employeeForm.querySelector('[data-qa="age"]');
+  const salaryInput = employeeForm.querySelector('[data-qa="salary"]');
+
+  function isNameOrPosition(value) {
+    const trimValue = value.trim();
+    const regex = /^[a-zA-Z\s]+$/;
+
+    return regex.test(trimValue);
+  }
+
+  function isValidName(value) {
+    if (
+      typeof value === 'string' &&
+      value.trim().length >= 4 &&
+      isNameOrPosition(value)
+    ) {
+      return true;
+    } else {
+      return false;
+    }
+  }
+
+  function isValidPosition(value) {
+    if (
+      typeof value === 'string' &&
+      value.trim().length >= 1 &&
+      isNameOrPosition(value)
+    ) {
+      return true;
+    } else {
+      return false;
+    }
+  }
+
+  function isValidAge(value) {
+    if (!isNaN(Number(value)) && value >= 18 && value <= 90) {
+      return true;
+    } else {
+      return false;
+    }
+  }
+
+  function isValidSalary(value) {
+    if (!isNaN(Number(value))) {
+      return true;
```

Comment:
The `isValidSalary` function currently only checks if the value can be converted to a number. It should also ensure that the salary is in a valid format, such as including a dollar sign and commas where appropriate.